airflow/contrib/hooks/aws_athena_hook.py

```diff
@@ -0,0 +1,147 @@
+            self.log.info("Query is in {state} state. Cannot fetch results".format(state=state))
+            return None
+        return self.conn.get_query_results(QueryExecutionId=self._queryExecutionId)
+
+    def _stop_query(self):
+        if self._queryExecutionId is None:
+            return None
+        return self.conn.stop_query_execution(QueryExecutionId=self._queryExecutionId)
+
+    def _poll_query_status(self, sleep_time=None, max_tries=None):
+        try_number = 1
+        if sleep_time is None:
+            sleep_time = self.sleep_time
+        if max_tries is None:  # Retries forever until query reaches final state
+            retry_condition = True
+        else:
+            retry_condition = try_number <= max_tries
+        while retry_condition:
+            response = self._check_query_status()
+            state = None
+            try:
+                state = response['QueryExecution']['Status']['State']
+            except Exception as ex:
+                self.log.error("{try_number}. Error occurred while fetching query state".format(
+                    try_number=try_number), ex)
+            finally:
+                if state is None:
+                    self.log.error("Trial {try_number}: Invalid query state. Retrying again".format(
+                        try_number=try_number))
+                elif state in INTERMEDIATE_STATES:
+                    self.log.info("Trial {try_number}: Query is still in an intermediate state - {state}"
+                                  .format(try_number=try_number, state=state))
+                else:
+                    self.log.info("Trial {try_number}: Query execution completed. Final state is {state}"
+                                  .format(try_number=try_number, state=state))
+                    break
+                try_number += 1
+                if max_tries:
+                    retry_condition = try_number <= max_tries
+                sleep(sleep_time)
+
+    def get_conn(self):
+        self.conn = self.get_client_type('athena', self.region_name)
+        return self.conn
+
+    def run_query(self):
+        response = self.conn.start_query_execution(QueryString=self.query,
+                                                   ClientRequestToken=self.client_request_token,
+                                                   QueryExecutionContext=self.query_execution_context,
+                                                   ResultConfiguration=self.result_configuration)
+        self._queryExecutionId = response['QueryExecutionId']
+        return response
+
+    def check_query_status(self):
+        response = self._check_query_status()
+        state = None
+        try:
+            state = response['QueryExecution']['Status']['State']
+        finally:
+            return state
+
+    def poll_query_status(self, sleep_time=None, max_tries=None):
+        self._poll_query_status(sleep_time=None, max_tries=max_tries)
+
+    def on_kill(self):
```

Comment:
On kill does not belong in the hook.

All of this logic needs to be in the operator.